### PR TITLE
Added quote to import_table.

### PR DESCRIPTION
**Pull request by cseed**
Needed for UKBB phenotype import.

### CODE REVIEW

File: python/hail/context.py [hunk 1]
```diff
@@ -288,10 +288,11 @@ def import_gen(self, path, sample_file=None, tolerance=0.2, min_partitions=None,
                       no_header=bool,
                       comment=nullable(strlike),
                       delimiter=strlike,
+                      quote=nullable(char),
                       missing=strlike,
                       types=dictof(strlike, Type))
     def import_table(self, paths, key=[], min_partitions=None, impute=False, no_header=False,
-                     comment=None, delimiter="\t", missing="NA", types={}):
+                     comment=None, delimiter="\t", quote=None, missing="NA", types={}):
```
Comment (tpoterba) [inline python/hail/context.py]: breaking change. it has to go at the end

File: python/hail/typecheck/check.py [hunk 2]
```diff
@@ -131,6 +140,7 @@ def dictof(k, v):
 
 numeric = oneof(int, long, float)
 
+char = CharChecker()
```
Comment (tpoterba) [inline python/hail/typecheck/check.py]: need to add this to the init

File: src/main/scala/is/hail/HailContext.scala [hunk 3]
```diff
@@ -312,21 +312,23 @@ class HailContext private(val sc: SparkContext,
     types: java.util.HashMap[String, Type],
     commentChar: String,
     separator: String,
+    quote: java.lang.Character,
```
Comment (tpoterba) [inline src/main/scala/is/hail/HailContext.scala]: won't it be a string coming in?

File: src/main/scala/is/hail/utils/TextTableReader.scala [hunk 4]
```diff
@@ -11,23 +11,81 @@ import org.apache.spark.sql.Row
...
+        while (i < s.length && s(i) != quote) {
+          sb += s(i)
+          i += 1
+        }
+
+        if (i == s.length)
+          fatal(s"missing terminating quote character $quote")
+        i += 1 // skip quote
+
+        // full field must be quoted
```
Comment (tpoterba) [inline src/main/scala/is/hail/utils/TextTableReader.scala]: comments? from cotton seed? 😱

File: src/main/scala/is/hail/HailContext.scala (hunk 3 above)
Comment (cseed) [inline src/main/scala/is/hail/HailContext.scala]: If the Java side is char, py4j passes the first character of the python side.  If the string is empty on the python side, py4j throws `StringIndexOutOfBoundsException`.

**Comment by cseed** [comment]
Addressed comments.

**Pull request merged**